deployments/rialto/dashboard/grafana/provisioning/dashboards/grafana-dashboard.json

```diff
@@ -18,66 +18,143 @@
   "links": [],
   "panels": [
     {
+      "alert": {
+        "alertRuleTags": {},
+        "conditions": [
+          {
+            "evaluator": {
+              "params": [
+                5
+              ],
+              "type": "gt"
+            },
+            "operator": {
+              "type": "and"
+            },
+            "query": {
+              "params": [
+                "A",
+                "5m",
+                "now"
+              ]
+            },
+            "reducer": {
+              "params": [],
+              "type": "max"
+            },
+            "type": "query"
+          }
+        ],
+        "executionErrorState": "alerting",
+        "for": "5m",
+        "frequency": "5m",
+        "handler": 1,
+        "message": "",
+        "name": "Rialto Block Difference > 5",
+        "noDataState": "no_data",
+        "notifications": []
+      },
+      "aliasColors": {},
+      "bars": false,
+      "dashLength": 10,
+      "dashes": false,
       "datasource": "Prometheus",
+      "description": "Difference between source & target blocks",
       "fieldConfig": {
         "defaults": {
-          "custom": {
-            "align": null
-          },
-          "mappings": [],
-          "thresholds": {
-            "mode": "absolute",
-            "steps": [
-              {
-                "color": "green",
-                "value": null
-              },
-              {
-                "color": "red",
-                "value": 80
-              }
-            ]
-          }
+          "custom": {}
         },
         "overrides": []
       },
+      "fill": 1,
+      "fillGradient": 0,
       "gridPos": {
-        "h": 6,
+        "h": 8,
         "w": 12,
         "x": 0,
         "y": 0
       },
-      "id": 2,
-      "interval": "5s",
-      "options": {
-        "colorMode": "value",
-        "graphMode": "area",
-        "justifyMode": "auto",
-        "orientation": "auto",
-        "reduceOptions": {
-          "calcs": [
-            "mean"
-          ],
-          "fields": "",
-          "values": false
-        }
+      "hiddenSeries": false,
+      "id": 14,
+      "legend": {
+        "avg": false,
+        "current": false,
+        "max": false,
+        "min": false,
+        "show": true,
+        "total": false,
+        "values": false
       },
-      "pluginVersion": "7.0.6",
+      "lines": true,
+      "linewidth": 1,
+      "nullPointMode": "null",
+      "percentage": false,
+      "pluginVersion": "7.1.0",
+      "pointradius": 2,
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
       "targets": [
         {
-          "expr": "best_block_numbers",
-          "format": "time_series",
-          "instant": true,
+          "expr": "max(best_block_numbers{node=\"source\"}) - max(best_block_numbers{node=\"target\"})",
```

Comment:
Technically we may not receive alerts even if sync is not working, because both metrics are coming from relay => relay may be offline && report the same (<5) difference again and again. But we have nothing better now.

Except, maybe additional alerts if 'max(best_block_numbers{node=\"source\"})' or `max(best_block_numbers{node=\"target\"})` aren't changing for too long. But here comes additional relay feature - it doesn't ask source node (PoA in eth2sub mode) about best block if the difference is too big. Because we don't actually need this information until we have synchronized almost everything. Example: we start Kovan -> Rialto sync. Kovan has almost 20_000_000 headers. Let's say it'll take 4 days to sync this chain (this just an example - not an estimation). So for all this time, `max(best_block_numbers{node=\"source\"})` will be 20_000_000 even though another 1_000_000 may have been generated on Kovan. This is just a small optimization (because we do not need this information for headers synchronization) and can be eliminated if required.

Reply:
IMHO Initial sync is a special case anyway, we should probably temporarily turn off alerts during that time anyway. I like the idea of adding an alert if there is no progress in PoA blocks. Will do.